applications/StructuralMechanicsApplication/custom_elements/prestress_membrane_element.cpp

```diff
@@ -671,25 +588,31 @@ void PrestressMembraneElement::CalculateAll(
     const unsigned int number_of_nodes = GetGeometry().size();
     const unsigned int mat_size = number_of_nodes * 3;
 
-    // Matrix B(3, mat_size);
-    // change to: array_1d<double, 3> StrainVector;
-    Vector StrainVector(3);
-    Vector StressVector(3);
+    Vector strain_vector(3);
+    Vector stress_vector(3);
 
     // set up Constitutive Law
     ConstitutiveLaw::Parameters Values(GetGeometry(), GetProperties(), rCurrentProcessInfo);
 
     // Set constitutive law flags:
-    Flags &ConstitutiveLawOptions=Values.GetOptions();
-    ConstitutiveLawOptions.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, false);
-    ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_STRESS, true);
-    ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, true);
+    Flags &constitutive_law_options=Values.GetOptions();
+    constitutive_law_options.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, true);
+    constitutive_law_options.Set(ConstitutiveLaw::COMPUTE_STRESS, true);
+    // for formfinding: Constitutive Tensor is 0 Tensor
+    if(this->Has(IS_FORMFINDING)){
```

Comment:
you can set to true by default and then use a simplified if-construct

Reply:
done